tuna.go

```diff
@@ -438,21 +449,22 @@ func (c *Common) wrapConn(conn net.Conn, remotePublicKey []byte, localConnMetada
 	defer conn.SetDeadline(time.Time{})
 
 	if len(remotePublicKey) > 0 {
+		remoteConnMetadata, err = readConnMetadata(conn)
+		if err != nil {
+			return nil, nil, err
+		}
+
+		connNonce = remoteConnMetadata.Nonce
 		encryptionAlgo = c.encryptionAlgo
 		localConnMetadata.EncryptionAlgo = encryptionAlgo
 		localConnMetadata.PublicKey = c.Wallet.PubKey()
+		localConnMetadata.Nonce = connNonce
```

Comment:
this will make nonce to be sent back to server side

Reply:
We need  to return `nonce` as  part of key to remove encrypt key when connection closed

> this will make nonce to be sent back to server side